pkg/csr/csr.go

```diff
@@ -86,18 +88,24 @@ func (a *Approver) Approve() error {
 		return errors.Wrapf(err, "could not validate contents for approval of CSR: %s", a.csr.Name)
 	}
 
-	a.csr.Status.Conditions = append(a.csr.Status.Conditions, certificates.CertificateSigningRequestCondition{
+	approvalCond := certificates.CertificateSigningRequestCondition{
 		Type:           certificates.CertificateApproved,
 		Status:         "True",
-		Message:        "Approved by WMCO",
+		Message:        "This CSR was approved by the WMCO certificate Approver.",
 		LastUpdateTime: meta.Now(),
-		Reason:         "WMCOApproved",
-	})
-
-	if _, err := a.k8sclientset.CertificatesV1().CertificateSigningRequests().UpdateApproval(context.Background(),
```

Comment:
in most cases i'd think patching would be the way to go here, but considering there is an explicit `UpdateApproval()` method in the client, is that not the recommended way to approve a CSR?

Can you find any docs that support switching to the patch method?

Reply:
Approval is a subresource of the CSR, so just as other resources have an `Update()` method upon them, CSRs have `Update()`, `UpdateApproval()`, and `Update<OtherSubresource>()`. Under the hood they are all just simple HTTP Put calls. Note they also all have `Patch()`.

In my mind this switch has the same justification as other switches to Patch over Update (e.g. https://github.com/openshift/windows-machine-config-operator/pull/518). 
`A patch is best used when
modifying parts of a resource, which fits the CSR approval case;
this is why a patch was favored over a Get -> Update loop.`

Reply:
The recommendation in the kubernetes docs is to use update 
https://v1-18.docs.kubernetes.io/docs/reference/access-authn-authz/certificate-signing-requests/#authorization